pydm/data_plugins/local_plugin.py

```diff
@@ -1,332 +1,427 @@
-"""
-Plugin to allow users to arbitrarily connect python object attributes to widget
-channels. This handles the polling tasks to update the gui.
-"""
-import inspect
+"""Local Plugin."""
+import decimal
+import logging
+from urllib import parse
+import shlex
 import numpy as np
-from qtpy.QtWidgets import QWidget
-from qtpy.QtCore import Slot, Qt, QCoreApplication, QTimer
+from qtpy.QtCore import Slot, Qt
 from pydm.data_plugins.plugin import PyDMPlugin, PyDMConnection
 
+logger = logging.getLogger(__name__)
+
+class Connection(PyDMConnection):
+    def __init__(self, channel, address, protocol=None, parent=None):
+        self._is_connection_configured = False
+        self._value_type = None
+        self._precision_set = None
+        self._type_kwargs = {}
+
+        self._required_config_keys = [
+            'name',
+            'type',
+            'init'
+        ]
+
+        self._extra_config_keys = [
+            "precision",
+            "unit",
+            "upper_limit",
+            "lower_limit",
+            "enum_string"
+            ]
+
+        self._data_types = {
+            'int': int,
+            'float': float,
+            'str': str,
+            'bool': bool,
+            'array': np.array
+            }
+
+        self._precision = None
+        self._unit = None
+        self._upper_limit = None
+        self._lower_limit = None
+        self._enum_string = None
+
+        super(Connection, self).__init__(channel, address, protocol, parent)
+        self.add_listener(channel)
+        self._configuration = {}
+
+        self.send_connection_state(False)
+        self.send_access_state()
+
+        self.connected = False
+
+    def _configure_local_plugin(self, channel):
+        if self._is_connection_configured:
+            logger.debug('LocalPlugin connection already configured.')
+            return
 
-class LocalPlugin(PyDMPlugin):
-    """
-    Plugin that only exists in a local space to connect to a local object.
+        try:
+            address = PyDMPlugin.get_address(channel)
+            address = "loc://" + address
+
+            self._configuration = parse.parse_qs(parse.urlsplit(address).query)
+            name = parse.urlsplit(address).netloc
+            self._configuration['name'] = name
+
+            if not name and not config:
+                raise
+        except:
```

Comment:
Bare except is usually bad practice because it catches errors like `KeyboardInterrupt`
```suggestion
        except Exception:
```